### DISCUSSION TITLE
Visibility settings for features

### DISCUSSION THREAD

**Discussion by tswfi** [PrestaShop/PrestaShop#40177] [Ideas]
#### Is your feature request related to a problem?

As a merchant I sometimes want to save some extra information to products just for me.

Features is a nice way to add information to products but they are always visible for customers also.

#### Describe the solution you'd like

Setting for feature so that is not shown in storefront but only as a extra information for the merchant.
Perhaps a setting if the feature is searchable also.

#### Alternatives you've considered

Writing a new module just to get one more information saved for a product is kinda tedious.

#### Additional context

It would also be nice if features could have "types" like this feature is a html field and this feature is always predefined etc. This way module developers could lean on core functionality for adding new information for products.

If the features would also have clean import export functionality for products this would make keeping product data nice and clean inside the core and modules could then just use existing core functionality easily.

**Comment by florine2623** [comment]
Hello @tswfi ,

Thanks for the suggestion.
Could you also provide a mock-up of what it would look like ?
Thanks!

**Comment by tswfi** [comment]
![image](https://user-images.githubusercontent.com/1352132/110793454-4ed46f80-827d-11eb-96b9-9ccc2dd7a4b8.png)

but with the nice `x` and `checkmark` icons

![image](https://user-images.githubusercontent.com/1352132/110793675-93600b00-827d-11eb-8b58-527f85608460.png)

normal settings form for the feature.


And if type is "html textarea" the product editing:
![image](https://user-images.githubusercontent.com/1352132/110794064-ffdb0a00-827d-11eb-8b3a-1af2e67f0c41.png)

but without the copy paste errors I just did while editing the dom directly.

And the feature could also have a setting "show as a new tab". This way the feature could have something like food ingredients list etc.

**Comment by florine2623** [comment]
Ping @PrestaShop/product-team ,

@tswfi is suggesting a new feature to add to the BO, what do you think ?

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @tswfi 

I think what you are suggesting is this:

![image](https://user-images.githubusercontent.com/23308427/110981981-40af4d80-8368-11eb-9782-12e20ad0bdd6.png)

I don't know if it's needed but I think you can add it yourself by overriding the features :wink: 
Well let's see the point of view of the product team, stay tuned!

Ping @PrestaShop/product-team @tswfi suggests adding a new attribute for the feature object **Status** that will be helpful to the merchants, wdyt?

**Comment by tswfi** [comment]
Term "status" sounds more like totally on/off.
"Visibility in SF" is the setting I'm after.
Yes I am aware that overriding or writing totally new feature is a possibilty, but I think adding new attributes to products is pretty basic core functionality for a webshop framework.

**Comment by MatShir** [comment]
@tswfi could you detail a bit why you would need this feature as a merchant ?
It looks an interesting suggestion, but it requires a bit more context to evaluate the feature ?

**Comment by tswfi** [comment]
@MatShir as a merchant:
* saving extra data under products that is useful for the merchant but should not be shown to customers
  * the merchant case where this came up in was with customs CN codes. Merchant needs these codes and a nice place to keep them with the product data would be in a product feature. But that information is not required for the customers to see.
* another case would be while enriching products with the extra data the merchant might want keep the field hidden until all products have been enriched (this might take a while if there are lots of products)

And as a module developer this is a bit similar to what I've talked here more in depth: https://github.com/PrestaShop/docs/issues/896#issuecomment-802007375
